.github/workflows/tests.yml

```diff
@@ -28,16 +28,86 @@ jobs:
     - uses: actions/checkout@v4
 
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v5
+      uses: astral-sh/setup-uv@v5
       with:
         python-version: ${{ matrix.python-version }}
 
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install -r requirements-dev.txt
-        pip install -e .
+    - name: Test with python ${{ matrix.python-version }}
+      run: uv run --frozen -m pytest
+
+  build:
+    name: Build package
+    runs-on: ubuntu-latest
+    needs: [test]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install the latest version of uv and set the python version
+        uses: astral-sh/setup-uv@v5
+        with:
+          python-version: 3.12
+      - name: Build
+        run: uv build
+      - name: Store the distribution packages
+        uses: actions/upload-artifact@v4
+        with:
+          name: python-package-distributions
+          path: dist/
+
+  publish-to-pypi:
+    name: Publish to PyPI
+    runs-on: ubuntu-latest
+    # if: "startsWith(github.ref, 'refs/tags/')"
```

Comment:
Uncomment this before merging.

You can use [TestPyPI](https://packaging.python.org/en/latest/guides/using-testpypi/) for testing. I often publish to TestPyPI first to make sure everything is working ok.